File: Java/src/main/java/com/gildedrose/SulfurasUpdater.java
package com.gildedrose;

public class SulfurasUpdater extends ItemUpdater {

    @Override
    void updateQuality(Item item) {
        System.out.println("########Sulfuras is a legendary product");
        // TODO
        item.quality = 80;
    }

    @Override
    void updateSellIn(Item item) {
        System.out.println("########Never gets old");
    }

    @Override
    boolean canUpdateQuality(Item item) {
        // "Sulfuras", being a legendary item, never decreases in Quality
        return false;
    }

    @Override
    int getDegradeValue(Item item) {
        // "Sulfuras", being a legendary item, never has to be sold
        return 0;
    }

    public int getHighestQuality() {
        return 80;
    }
}

File: js-jest/src/item_updaters/sulfuras_updater.js
const {ItemUpdater} = require("../gilded_rose");

class SulfurasUpdater extends ItemUpdater {
  updateQuality() {
    
    //The Quality is 80 and it never alters
    if (this.item.quality !== 80) {
      this.item.quality = 80;
    }
    return this.item;
  }
  }

  module.exports = {
    SulfurasUpdater
  }